dev/doc/xml-protocol.md

```diff
@@ -194,8 +193,9 @@ Moves current tip to `${stateId}`, such that commands may be added to the new st
 ```
 * Failure: If `stateId` is in an error-state and cannot be jumped to, `errorFreeStateId` is the parent state of `stateId` that should be edited instead.
 ```html
-<value val="fail" loc_s="${startOffsetOfError}" loc_e="${endOffsetOfError}">
+<value val="fail">
   <state_id val="${errorFreeStateId}"/>
+  <option val="none|some">${loc}</option>
   ${errorMessage}
 </value>
 ```
```

Comment:
The description at line 924 ("* <a name="location">Location</a>, a Coq location:") omits the fname field, which is part of Loc.t but not in the protocol (arguably it should be sent, too).  Probably it should also say that it's based on the `Loc.t` type (more specific).

Reply:
Good point. Maybe should we send the `fname` too? Are there situations where Coq reports about locations in a file rather than in the toplevel input?

Reply:
For reporting errors, warnings, etc. I'm pretty sure CoqIDE doesn't currently report anything that's outside the current buffer. The debugger "get stack" message includes the absolute filename with `bp/ep` offsets.  It will open a buffer for the referenced file if needed.  Let's not touch that--it works correctly now and doesn't have the location problems we're trying to address for error messages.  (Indeed, part of the problem with the error message locations has including unneeded, untested and buggy "clean up".)

Still not super keen about reusing `Loc.t` in CoqIDE.  It would be logical to receive a Loc.t in CoqIDE, then adjust all the fields in one place (encapsulation).  Putting the adjusted fields back into a Loc.t means that sometimes Loc.t has byte offsets and sometimes it has char offsets--easy to confuse and requires close reading of the code to follow.  Your choice.

Reply:
I'm not competent to judge here. Apparently, it works with `Loc.t`. And as far as I can judge, to work properly, it should *only* have byte offsets (as in Coq).